File: js/task1/S1_IF_03.js
// S1_IF_03 — Truthy/falsy guard

function normalizeName(input) {
  if (!input) {
    return "Anonymous";
  }

  const trimmed = input.trim();
  return trimmed ? trimmed : "Anonymous";
}

// Tests
console.log(normalizeName(""));        // Anonymous
console.log(normalizeName(" "));       // Anonymous
console.log(normalizeName(null));      // Anonymous
console.log(normalizeName(" Ola "));   // Ola

File: python/task1b/S1_IF_03.py
# S1_IF_03 — Normalize user name

def normalize_name(x):
    if not x:
        return "Anonymous"

    name = str(x).strip()
    return name if name else "Anonymous"


def main():
    tests = ["", " ", None, " Ola "]

    for t in tests:
        print(f"input={repr(t):>6} -> output={normalize_name(t)}")


if __name__ == "__main__":
    main()
